src/orange/g4org/SolidConverter.cc

```diff
@@ -336,8 +336,55 @@ auto SolidConverter::genericpolycone(arg_type solid_base) -> result_type
 auto SolidConverter::generictrap(arg_type solid_base) -> result_type
 {
     auto const& solid = dynamic_cast<G4GenericTrap const&>(solid_base);
-    CELER_DISCARD(solid);
-    CELER_NOT_IMPLEMENTED("generictrap");
+
+    auto const vtx = solid.GetVertices();
+    CELER_ASSERT(vtx.size() == 8);
+
+    std::vector<GenTrap::Real2> lower(4), upper(4);
+    for (auto i : range(4))
+    {
+        lower[i] = scale_.to<Cone::Real2>(vtx[i].x(), vtx[i].y());
+        upper[i] = scale_.to<Cone::Real2>(vtx[i + 4].x(), vtx[i + 4].y());
+    }
+    real_type hh = scale_(solid.GetZHalfLength());
+
+    return make_shape<GenTrap>(solid, hh, lower, upper);
+}
+
+//---------------------------------------------------------------------------//
+
+//! Convert a trapezoid
+auto SolidConverter::trap(arg_type solid_base) -> result_type
+{
+    auto const& solid = dynamic_cast<G4Trap const&>(solid_base);
+
+    real_type sinphi, cosphi;
+    sincos(solid.GetPhi(), &sinphi, &cosphi);
+    auto tanTheta = std::tan(solid.GetTheta());
```

Comment:
Regarding the build error: see https://github.com/celeritas-project/celeritas/pull/830 . I think for now (until we discuss https://github.com/celeritas-project/celeritas/issues/1132 ) we can just apply @drbenmorgan 's fix to this piece of code.

Also please follow the style guide and use `tan_theta`.